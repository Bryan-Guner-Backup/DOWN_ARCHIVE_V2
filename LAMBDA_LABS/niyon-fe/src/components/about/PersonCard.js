import React from 'react'
import teamMembers from './TeamMembers'

const PersonCard = (props) => {
  return (
    <div className="person" data-test="person-container">
      {teamMembers.map((teamMember, index) => {
        return (
          <div key={index} data-test="person-image-container">
            <a href={teamMember.link}>
              <img
                className="img2, personImg"
                src={teamMember.src}
                alt="Niyon Team Members"
              />
            </a>
            <h2>{teamMember.name}</h2>
            <p>{teamMember.position}</p>
          </div>
        )
      })}
    </div>
  )
}

export default PersonCard
