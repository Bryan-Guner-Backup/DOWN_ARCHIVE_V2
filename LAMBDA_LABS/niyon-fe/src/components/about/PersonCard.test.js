import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import PersonCard from './PersonCard'
import teamMembers from './TeamMembers'
import findByTestAttr from '../../tests/utils'

const setUp = (props = {}) => {
  const component = shallow(<PersonCard { ...props } />)
  return component
}

describe('<PersonCard /> component testing', () => {
  let component
  const testProps = {
    name: 'test-name',
    position: 'test-position'
  }
  beforeEach(() => {
    component = setUp(testProps)
  })

  it('should pass the snapshot test correctly', () => {
    expect(toJSON(component)).toMatchSnapshot()
  })

  it('should render <PersonCard /> component with no errors', () => {
    expect(component.length).toBe(1)
  })

  it('should render PersonCard container', () => {
    const personContainer = findByTestAttr(component, 'person-container')
    expect(personContainer.length).toBe(1)
  })

  describe('PersonCard Image container', () => {
    let personCard
    beforeEach(() => {
      personCard = findByTestAttr(component, 'person-image-container')
    })
    it('should render PersonCard Image container', () => {
      personCard.forEach((div) => {
        expect(div.length).toBe(1)
      })
    })

    it('should render anchor tag for image link', () => {
      const a = component.find('a')
      expect(a.length).toBe(teamMembers.length)
    })

    it('should render <h2 /> correctly', () => {
      const header = component.find('h2')
      expect(header.exists()).toBe(true)
    })

    it('should render header content correctly', () => {
      personCard.forEach((div, index) => {
        expect(div.find('h2').text()).toEqual(teamMembers[index].name)
      })
    })

    it('should render <p/> correctly with its contents', () => {
      personCard.forEach((div, index) => {
        expect(div.find('p').text()).toEqual(teamMembers[index].position)
      })
    })
  })
})
