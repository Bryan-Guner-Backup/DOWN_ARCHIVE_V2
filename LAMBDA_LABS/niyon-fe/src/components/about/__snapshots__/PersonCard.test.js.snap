// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonCard /> component testing should pass the snapshot test correctly 1`] = `
<div
  className="person"
  data-test="person-container"
>
  <div
    data-test="person-image-container"
    key="0"
  >
    <a
      href="https://github.com/carlostbanks"
    >
      <img
        alt="Niyon Team Members"
        className="img2, personImg"
        src="https://avatars3.githubusercontent.com/u/48330194?s=460&u=a27fc688702a28e9c13cba3a76cd8c324ec754c5&v=4"
      />
    </a>
    <h2>
      Carlos Banks
    </h2>
    <p>
      Fullstack Developer
    </p>
  </div>
  <div
    data-test="person-image-container"
    key="1"
  >
    <a
      href="https://github.com/wombat1972"
    >
      <img
        alt="Niyon Team Members"
        className="img2, personImg"
        src="https://avatars3.githubusercontent.com/u/53923403?s=460&v=4"
      />
    </a>
    <h2>
      Kenneth Brook
    </h2>
    <p>
      Fullstack Developer
    </p>
  </div>
  <div
    data-test="person-image-container"
    key="2"
  >
    <a
      href="https://github.com/farjadfazli"
    >
      <img
        alt="Niyon Team Members"
        className="img2, personImg"
        src="https://avatars3.githubusercontent.com/u/51179586?s=460&u=e8574ff7f2ad8da4183ed14e3571a1aa96a425f0&v=4"
      />
    </a>
    <h2>
      Farjad Fazli
    </h2>
    <p>
      Fullstack Developer
    </p>
  </div>
  <div
    data-test="person-image-container"
    key="3"
  >
    <a
      href="https://github.com/TimMitzen"
    >
      <img
        alt="Niyon Team Members"
        className="img2, personImg"
        src="https://ca.slack-edge.com/ESZCHB482-W012JHR9Y4A-a5a857f545bd-512"
      />
    </a>
    <h2>
      Tim Mitzen
    </h2>
    <p>
      Fullstack Developer
    </p>
  </div>
  <div
    data-test="person-image-container"
    key="4"
  >
    <a
      href="https://github.com/Joe-Thompson"
    >
      <img
        alt="Niyon Team Members"
        className="img2, personImg"
        src="https://avatars1.githubusercontent.com/u/51089982?s=460&u=9a11171c2d1647d5b68ad5754f42d1c4d8f62d84&v=4"
      />
    </a>
    <h2>
      Joe Thompson
    </h2>
    <p>
      Fullstack Developer
    </p>
  </div>
  <div
    data-test="person-image-container"
    key="5"
  >
    <a
      href="https://github.com/vy3191"
    >
      <img
        alt="Niyon Team Members"
        className="img2, personImg"
        src="https://avatars3.githubusercontent.com/u/26292705?s=460&u=2611df50ac310da56f959adb32bf6c4bb40b2d69&v=4"
      />
    </a>
    <h2>
      Venky Yagatilee
    </h2>
    <p>
      Fullstack Developer
    </p>
  </div>
  <div
    data-test="person-image-container"
    key="6"
  >
    <a
      href="https://github.com/jeremyRogel"
    >
      <img
        alt="Niyon Team Members"
        className="img2, personImg"
        src="https://avatars3.githubusercontent.com/u/52545325?s=460&u=390fe26e6eb18db20122f3239890c3b0a6722b80&v=4"
      />
    </a>
    <h2>
      Jeremy Rogel
    </h2>
    <p>
      Team Lead (RC2)
    </p>
  </div>
  <div
    data-test="person-image-container"
    key="7"
  >
    <a
      href="https://github.com/CarlosVillalpandoJr"
    >
      <img
        alt="Niyon Team Members"
        className="img2, personImg"
        src="https://avatars3.githubusercontent.com/u/29393301?s=460&u=33d6fa041db14a6caaaa1d44ad2640a71cea2c31&v=4"
      />
    </a>
    <h2>
      Carlos Villalpando
    </h2>
    <p>
      Team Lead (RC1)
    </p>
  </div>
</div>
`;
